### PR TITLE
[ASR][V2A-RCM] Add properties and provider specific models failback provider for V2A RCM.

### PR DESCRIPTION
**Pull request by imshashwataggarwal**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [ ] new API version 
      - [X] update existing version for new feature 
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet.
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet.
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [X] SDK of .NET (need service team to ensure code readiness)
      - [X] SDK of Python
      - [X] SDK of Java
      - [X] SDK of Js
      - [X] SDK of Go
      - [X] PowerShell
      - [X] CLI
      - [X] Terraform
      - [ ] No, no need to refresh for updates in this PR

### Contribution checklist:
- [X] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of “no breaking changes
- [X] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [X] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [X] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [X] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from API Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable version with Breaking Change Validation errors
- [ ] Updating API(s) in preview over 1 year

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://msazure.visualstudio.com/One/_workitems/create/Scenario?templateId=d1c215ec-5d6a-4cb3-a193-95ddd0a1ba96&ownerId=85bb92fb-21fe-41ca-9309-47f2c5f57515). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

-----------------

This PR adds V2A-RCM specific request-response models and properties for failback direction. Two new PUT APIs and there corresponding examples are added to support "Cancel Failover" operation.
- /Subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.RecoveryServices/vaults/{resourceName}/replicationFabrics/{fabricName}/replicationProtectionContainers/{protectionContainerName}/replicationProtectedItems/{replicatedProtectedItemName}/failoverCancel
- /Subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.RecoveryServices/vaults/{resourceName}/replicationRecoveryPlans/{recoveryPlanName}/failoverCancel

### CODE REVIEW

**Comment by imshashwataggarwal** [comment]
Adding @DevBaburaj, @anmolbhatia289

**Comment by imshashwataggarwal** [comment]
@ChenTanyi, Please review the PR.

**Comment by ChenTanyi** [comment]
> Hi, @imshashwataggarwal your PR are labelled with WaitForARMFeedback. A notification email will be sent out shortly afterwards to notify ARM review board([armapireview@microsoft.com](mailto:armapireview@microsoft.com)). cc @

@imshashwataggarwal Adding new API need ARM review first, please wait or ask ARM to review it.
@jianyexi Is adding a new path and new properties in current API not considered as breaking change? I saw the breaking change CI has passed.

**Comment by imshashwataggarwal** [comment]
@Azure/arm-api-review-board : Please review this.

File: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2018-07-10/service.json [hunk 1]
```diff
@@ -9049,6 +9165,10 @@
           "description": "The Id of the machine where the provider is getting added.",
           "type": "string"
         },
+        "biosId": {
```
Comment (pilor) [inline specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2018-07-10/service.json:9168]: This (along with all the other property changes in this PR) needs to go in a new api-version

File: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2018-07-10/service.json [hunk 2]
```diff
@@ -17187,6 +18191,34 @@
...
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "InMageRcmFailbackRecoveryPointType",
+            "modelAsString": true
+          }
+        },
+        "useMultiVmSyncPoint": {
+          "description": "A value indicating whether multi VM sync enabled VMs should use multi VM sync points for failover.",
+          "type": "string"
```
Comment (pilor) [inline specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2018-07-10/service.json:18217]: If this is contstrained to certain value please add the potential "enum" values with modelAsString: true

**Review by pilor — CHANGES_REQUESTED** [review]

File: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2018-07-10/service.json (hunk 1 above)
Comment (imshashwataggarwal) [reply specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2018-07-10/service.json:9168]: All the added properties are optional, so it shouldn't impact client usage and require a new api-version.

Comment (pilor) [reply specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2018-07-10/service.json:9168]: If you think this qualifies for an exception to the normal breaking change definition you can request an exemption via the process here:  https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes

Comment (mentat9) [reply specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2018-07-10/service.json:9168]: Adding optional properties is an evolutionary change that requires a new API version. Please see the MS API Guidelines here for details: https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md#api-changes-that-require-a-version-change

Comment (imshashwataggarwal) [reply specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2018-07-10/service.json:9168]: These properties are specific to V2A RCM provider which is not released to customers yet. So this shouldn't qualify as evolutionary change.

File: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2018-07-10/service.json (hunk 2 above)
Comment (imshashwataggarwal) [reply specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2018-07-10/service.json:18217]: this field can have true/false value only. It is modelled like other bool values in the spec.

Comment (pilor) [reply specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2018-07-10/service.json:18217]: A string whose only values are "true" or "false" should at the very least be a string enum.  If you are going to do this it should have been a boolean to begin with but a string enum will allow you to remain consistent with the other places that have a similar property while making it easy for users to use the SDK/API

File: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2018-07-10/service.json (hunk 1 above)
Comment (pilor) [reply specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2018-07-10/service.json:9168]: Please request an exception through the breaking change process linked above.  The swagger spec is for a stable api-version from 2018

**Pull request closed without merging**